tests/v2/test_peer_manager.nim

```diff
@@ -162,3 +163,53 @@ procSuite "Peer Manager":
     check:
       # Not currently connected to node2, but had recent, successful connection.
       node1.peerManager.connectedness(peerInfo2.peerId) == CanConnect
+    
+    await node2.stop()
+
+  asyncTest "Peer manager can use persistent storage and survive restarts":
+    let
+      database = SqliteDatabase.init("", inMemory = true)[]
+      storage = WakuPeerStorage.new(database)[]
+      nodeKey1 = crypto.PrivateKey.random(Secp256k1, rng[])[]
+      node1 = WakuNode.init(nodeKey1, ValidIpAddress.init("0.0.0.0"),
+        Port(60000), peerStorage = storage)
+      nodeKey2 = crypto.PrivateKey.random(Secp256k1, rng[])[]
+      node2 = WakuNode.init(nodeKey2, ValidIpAddress.init("0.0.0.0"),
+        Port(60002))
+      peerInfo2 = node2.peerInfo
+    
+    await node1.start()
+    await node2.start()
+
+    node1.mountRelay()
+    node2.mountRelay()
+
+    discard await node1.peerManager.dialPeer(peerInfo2, WakuRelayCodec, 2.seconds)
+    check:
+      # Currently connected to node2
+      node1.peerManager.peers().len == 1
+      node1.peerManager.peers().anyIt(it.peerId == peerInfo2.peerId)
+      node1.peerManager.connectedness(peerInfo2.peerId) == Connected
+
+    # Simulate restart by initialising a new node using the same storage
```

Comment:
Here a restart means stopping the node and then starting it? curious to know was there any issue with restarting node1 (as opposed to spinning a new node with the same storage)?

Reply:
No issue at all, but, since `node1` is already initialised in this scope, just stopping it won't destroy its peer manager and the persistent storage won't really be necessary. The easiest way to simulate a node being initialised "from scratch" is to just start a new one that loads from persistent storage.